drivers/counter/counter_mcux_rtc.c

```diff
@@ -236,6 +236,7 @@ static int mcux_rtc_init(const struct device *dev)
 	RTC_GetDefaultConfig(&rtc_config);
 	RTC_Init(config->base, &rtc_config);
 
+#if (defined(FSL_FEATURE_RTC_HAS_LPO_ADJUST) && FSL_FEATURE_RTC_HAS_LPO_ADJUST)
```

Comment:
This change makes no sense. The only thing this does it make the build assert not happen if `FSL_FEATURE_RTC_HAS_LPO_ADJUST` is not defined. But the point of the build assert is to check that if `FSL_FEATURE_RTC_HAS_LPO_ADJUST` is defined if `clock-source` in DT is `"LPO"`. So it makes the code dead. Just put the line `clock-source = "RTC"` in DT?

Reply:
then you can use IS_ENABLED to fix that https://docs.zephyrproject.org/latest/doxygen/html/group__sys-util.html#ga111fe4e9d63758262fc6810a782cb32a